__init__.py

```diff
@@ -778,98 +779,85 @@ async def get_directory_list(request):
+        async with session.get(url, headers=def_headers, allow_redirects=False) as rh:
+          if not rh.ok:
+            raise ValueError(
+              "Unable to download! Request header status code: " +
+              str(rh.status)
+            )
+
+          downloaded_size = 0
+          if rh.status == 200 and os.path.exists(filename_temp):
+            downloaded_size = os.path.getsize(filename_temp)
+
+          headers = {"Range": "bytes=%d-" % downloaded_size}
+          headers["User-Agent"] = def_headers["User-Agent"]
+          if "Authorization" in def_headers:
+            headers["Authorization"] = def_headers["Authorization"]
+
+          async with session.get(url, headers=headers, allow_redirects=False) as r:
+            if rh.status == 307 and r.status == 307:
+              # Civitai redirect
+              redirect_url = await r.text()
+              if not redirect_url.startswith("http"):
+                raise ValueError("Unable to download from Civitai! Redirect url: " + str(redirect_url))
+              await download_file(redirect_url, filename, overwrite)
+              return
+            if rh.status == 302 and r.status == 302:
+              # HuggingFace redirect
+              redirect_url = await r.text()
+              redirect_url_index = redirect_url.find("http")
+              if redirect_url_index == -1:
+                raise ValueError("Unable to download from HuggingFace! Redirect url: " + str(redirect_url))
+              await download_file(redirect_url[redirect_url_index:], filename, overwrite)
+              return
+            elif rh.status == 200 and r.status == 206:
+              # Civitai download link
+              pass
+
+            total_size = int(rh.headers.get("Content-Length", 0))
+
+            print("Downloading file: " + url)
+            if total_size != 0:
+              print("Download file size: " + str(total_size))
+
+            mode = "wb" if overwrite else "ab"
+            async with aiofiles.open(filename_temp, mode) as f:
+              async for chunk in r.content.iter_chunked(8192):
+                if chunk:
+                  downloaded_size += len(chunk)
+                  await f.write(chunk)
+
+                  if total_size != 0:
                     fraction = 1 if downloaded_size == total_size else downloaded_size / total_size
                     progress = int(50 * fraction)
                     sys.stdout.reconfigure(encoding="utf-8")
                     sys.stdout.write(
-                        "\r[%s%s] %d%%"
-                        % (
-                            "-" * progress,
-                            " " * (50 - progress),
-                            100 * fraction,
-                        )
+                      "\r[%s%s] %d%%"
+                      % (
+                        "-" * progress,
+                        " " * (50 - progress),
+                        100 * fraction,
+                      )
```

Comment:
This is kinda clunky while downloading files in parallel. Also it messes up the log file. I would suggest removing it and adding some other way to track download progress.

Reply:
@hayden-fr do you wanna keep this, or should I remove it?